packages/jades/src/verify.ts

```diff
@@ -0,0 +1,99 @@
+import { createVerify, X509Certificate } from 'node:crypto';
+import type { GeneralJWS } from './type';
+import { SDJwtGeneralJSONInstance } from '@sd-jwt/core';
+import { digest } from '@sd-jwt/crypto-nodejs';
+import { getGeneralJSONFromJWSToken } from './utils';
+
+export const JWTVerifier = {
+  async verify(credential: GeneralJWS | string, requiredClaimKeys?: string[]) {
+    const instance = new SDJwtGeneralJSONInstance({
+      hasher: digest,
+      verifier: JWTVerifier.verifier,
+    });
+
+    const generalJsonCredential = getGeneralJSONFromJWSToken(credential);
+
+    const verifiedData = await instance.verify(
+      generalJsonCredential,
+      requiredClaimKeys,
+    );
+    return verifiedData;
+  },
+
+  verifier(data: string, signatureB64: string): boolean {
+    try {
+      const [headerB64, payloadB64] = data.split('.');
+
+      const headerStr = Buffer.from(headerB64, 'base64url').toString('utf-8');
+      const header = JSON.parse(headerStr);
+
+      if (!header.x5c || !Array.isArray(header.x5c)) {
+        throw new Error('x5c certificate chain is missing in header');
+      }
+
+      const isValid = JWTVerifier.verifySig(
+        data,
+        signatureB64,
+        header.x5c,
+        header.alg,
+      );
+
+      return isValid;
+    } catch (error) {
+      console.error('JWT token verification error:', error);
```

Comment:
I would avoid to print errors to the console, it's just good for debugging, but not for production.